### ISSUE TITLE
Recurring donations are not showing as GIVbacks eligible :(

### ISSUE THREAD

**Issue by WhyldWanderer** [Giveth/impact-graph#1713]
It seems that from round 63 onward, the swagger data has not been including any recurring donations in the GIVbacks calculation.

63 - https://docs.google.com/spreadsheets/d/1iycPBLuoOD_Ck7ar9ErUfcghC6wASEnGfBpRAEtdRxI/edit?usp=sharing
64 - https://docs.google.com/spreadsheets/d/1prLUMWDVSv9kxiqWc-teLtnhs5GKhlLHBzxQl2dFUC0/edit?usp=sharing
65 - https://docs.google.com/spreadsheets/d/1sUUIxVrAPNyK4K-3MzXoQtLiEyVdWlmiRI3MJX949Ps/edit?usp=sharing
66 - https://docs.google.com/spreadsheets/d/1rjvSdwqv9f7f3GPxbyej5T1L4MxTYx-Q_2yxxn8WIwc/edit?usp=sharing

Please request access if needed.

**Comment by divine-comedian** [comment]
I have personally 4 different streams running to different verified projects on PRODUCTION - we can verify that there should be at least SOME givbacks eligible recurring donations that should appear..

![Screenshot from 2024-07-23 07-47-12](https://github.com/user-attachments/assets/dc752e5f-4121-446a-bd7d-962e9fb94038)


I'm not sure how @mohammadranjbarz setup the GIVbacks eligibility but I did notice that the streamable tokens have not been added to the adminJS panel for our token list.. I can't say if this was a requirement though since we do not want these streamable tokens to show on the one-time donation drop down list.

**Comment by kkatusic** [comment]
@divine-comedian, after @WhyldWanderer share me google sheet document we had call and went through all files and recurring donations were missing. We than generated new ones and seems that @mohammadranjbarz fixed this issue.
@WhyldWanderer will generate eligible donations again and check if there included all donation as it should and let you know.

**Comment by WhyldWanderer** [comment]
I have been doing some data diving and have found that the only round affected by this bug was round 62. All other rounds were distributed as expected.

33.86 worth of recurring donations did not receive GIVbacks in round 62

I have not received any complaints by users. Do you think that we should worry about trying to rectify R62 GIVbacks for such a small amount?

Im really not sure how I would do it, or if its worth it.

**Comment by WhyldWanderer** [comment]
The error must have occurred between R62 spreadsheet creation (June 3) and forum post creation (June 21)

This also means that the fix must have been implemented between June 3 and July 2

**Comment by divine-comedian** [comment]
> I have been doing some data diving and have found that the only round affected by this bug was round 62. All other rounds were distributed as expected.
> 
> 33.86 worth of recurring donations did not receive GIVbacks in round 62
> 
> I have not received any complaints by users. Do you think that we should worry about trying to rectify R62 GIVbacks for such a small amount?
> 
> Im really not sure how I would do it, or if its worth it.

We can distribute those GIVbacks but we can do a simpler method than the current GIVbacks process.. if you just DM me the address and amount of GIVbacks we can setup a manual distribution for R62

**Comment by WhyldWanderer** [comment]
> We can distribute those GIVbacks but we can do a simpler method than the current GIVbacks process.. if you just DM me the address and amount of GIVbacks we can setup a manual distribution for R62

I think the issue is the calculation part... Im not really sure how to calculate how much GIV they should get.
And we already sent out the 1M GIV that was allocated for Round 62

**Comment by laurenluz** [comment]
let's add the recurring donations we missed to the next round! as per GIVeconomy call

**Comment by WhyldWanderer** [comment]
Mitch mentioned in the chat in the call that we would spend more than 30 dollars manually adding these donations and then removing them again afterwards. Its actually quite a complicated task to manually add transactions when it comes to GIVbacks because they need to be reverted to the original data afterward... essentially the have to be added twice.

If we could agree on a percentage of GIVbacks for these donations and where the tokens would come from I think we can just distribute them manually without involving swagger api EVM crispr dev work etc.

**Comment by laurenluz** [comment]
maybe we just do a good average %? like 65%

**Comment by kkatusic** [comment]
@WhyldWanderer have we solved this?

**Comment by jainkrati** [comment]
@HrithikSampson and @kkatusic to update

**Comment by HrithikSampson** [comment]
@WhyldWanderer , Could I get access to the google spreadsheets.

**Comment by divine-comedian** [comment]
@HrithikSampson please follow with @WhyldWanderer on this issue

**Comment by HrithikSampson** [comment]
@WhyldWanderer , I am not sure what parameters and what endpoint to use to get the data for round 62 as you have shared in the excel sheet for optimism. I was checking the /calculated-updated as it only has Round Number as an input. How to get the data from givback-calculation to form the excel sheet.

https://docs.google.com/spreadsheets/d/19Ob_QRmBl2nQCSUR3f6YVNSm9pxft0WVtS6slC6E6Ek/edit?usp=sharing

**Comment by WhyldWanderer** [comment]
@HrithikSampson - the data is pulled from multiple APIs and I put it all together in one spreadsheet. 

The trouble with this issue is that the error has now been corrected. So, if we pull the data again, it wont really help us to solve this issue. Basically we need to find out which donations were missed. So we will have to look to see which donations are on the "eligible donations" list that are not on the "calculation list".

The issue I was facing is that when I tried to do the match, it did not add up. The total amount of recurring donations didnt make up for the discrepancy in the total amounts. Honestly, Im not sure its worth all of the time and resources to figure it out for $16 in missed donations. :/ No one has reported any issues with it.
